File: Scadex.Model/Dtos/ComponentTemplate/Queries/ComponentTemplatePalettePinDto.cs
using Scadex.Core.Model;
using static Scadex.Model.Enums.EntityEnums;

namespace Scadex.Model.Dtos.ComponentTemplate.Queries;

/// <summary>
/// Palet sablonunun pin semasindaki tek bir pin.
/// Buradaki alanlar tam olarak istemcinin cihazi canvas'ta cizmek ve pinlerini urettikten sonra kablo dogrulamasi yapmak icin ihtiyac duyduklaridir bilgilerdir.
/// Kaydedildikten sonra <see cref="Diagram.Queries.DiagramPinDto"/> olarak diyagramda kullanilir.
/// </summary>
public class ComponentTemplatePalettePinDto : IDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public double RelativeX { get; set; }
    public double RelativeY { get; set; }
    public HandleSide Side { get; set; }
    public PinFunction Function { get; set; }
    public PinDirection Direction { get; set; }
    public VoltageLevel? VoltageLevel { get; set; }
    public int? ChannelNumber { get; set; }
}

File: Scadex.WebUI/src/models/componentTemplate/queries/componentTemplatePalettePinDto.ts
/** Ayna: Scadex.Model/Dtos/ComponentTemplate/Queries/ComponentTemplatePalettePinDto.cs — sözleşme: docs/api-contract/10-component-template.md */
import type { HandleSide, PinDirection, PinFunction, VoltageLevel } from '@/models/enums';

/**
 * Palet şablonunun pin şemasındaki tek bir pin.
 *
 * Cihaz bırakılırken istemci bu şemadan gerçek pinleri üretir
 * (`lib/diagram/instantiate-template-pins.ts`): her pin için yeni bir Guid doğar,
 * geri kalan alanlar buradan kopyalanır ve aynı kopyalamayı sunucu da kendi
 * tarafında yapar.
 */
export interface ComponentTemplatePalettePinDto {
  /**
   * ŞABLON pininin Id'si — cihaz pininin Id'si değildir. Üretilen pin bunu
   * `componentTemplatePinId` olarak taşır ve gönderide sunucuya bu eşleme gider.
   */
  id: string;
  name: string;
  /** Şablonun genişliğine göre 0..1 normalize kesir. */
  relativeX: number;
  /** Şablonun yüksekliğine göre 0..1 normalize kesir. */
  relativeY: number;
  side: HandleSide;
  function: PinFunction;
  direction: PinDirection;
  /** Null = belirtilmemiş. Bağlantı doğrulaması yalnızca İKİ UÇ da doluysa karşılaştırır. */
  voltageLevel: VoltageLevel | null;
  /** Null = bu pinin telemetri kanalı yok; dolu olanlar cihazda `IoChannel` üretir. */
  channelNumber: number | null;
}
